### PR TITLE
Use `Circuit` for Bittide Nodes

### PR DESCRIPTION
**Pull request by lmbollen**
First effort to use `Circuit` in Bittide nodes

### CODE REVIEW

File: bittide/src/Bittide/Switch.hs
Comment (cuddlefishie) [inline bittide/src/Bittide/Switch.hs:40]:
```suggestion
  Circuit
    ( CSignal dom (Vec links (DataLink frameWidth))
    , Wishbone dom 'Standard addrW (Bytes nBytes) -- calendar interface
    )
```

File: bittide/src/Bittide/Node.hs
Comment (cuddlefishie) [inline bittide/src/Bittide/Node.hs:235]: Is this not compiling yet? I'll give it a try locally, just commenting here so I don't forget

**Review by cuddlefishie — COMMENTED** [review]
Overall looks good, could use a rebase since the Link stuff was removed and some comments should be updated.

File: bittide/src/Bittide/Node.hs
Comment (lmbollen) [reply bittide/src/Bittide/Node.hs:30]: https://github.com/clash-lang/clash-vexriscv/pull/36 :)

File: bittide/src/Bittide/Node.hs
Comment (hydrolarus) [inline bittide/src/Bittide/Node.hs:242]: The PR on `clash-protocols` is still in draft, should that be merged in before this PR or do we merge this first, given this PR is open?

**Review by hydrolarus — APPROVED** [review]
Looking good! Just a question about PR order stuff

File: bittide/src/Bittide/Node.hs
Comment (lmbollen) [reply bittide/src/Bittide/Node.hs:242]: Let's not block this PR on that, there's a tracking issue and I _intend_ to get back to that in the near future :)

**Pull request merged**
